daemon/graphdriver/devmapper/deviceset.go

```diff
@@ -2698,11 +2726,55 @@ func NewDeviceSet(root string, doInit bool, options []string, uidMaps, gidMaps [
 				return nil, err
 			}
 			devices.xfsNospaceRetries = val
+		case "dm.directlvm_device":
+			lvmSetupConfig.Device = val
```

Comment:
is it worth doing any validation that this is an actual device on the system? Obviously `pvcreate` will error out in some way if this is invalid, but maybe safer to do some simple validation. What does `pvcreate` do if I point it at my already mounted disk in some other filesystem format? I assume the error cases are "safe" but worth getting @rhvgoyal input on this..